kernel/src/fs/ext2/block_group.rs

```diff
@@ -55,7 +56,10 @@ impl BlockGroup {
                     }
                     let mut buf = vec![0u8; BLOCK_SIZE];
                     block_device.read_bytes(bid as usize * BLOCK_SIZE, &mut buf)?;
-                    Ok(IdAlloc::from_bytes_with_capacity(&buf, capacity))
+                    Ok(IdAlloc::from_bytes_with_capacity(
+                        &buf,
+                        capacity.align_up(SECTOR_SIZE),
```

Comment:
This is incorrect. The units of `capacity` and `SECTOR_SIZE` do not match: `capacity` is in bits, `SECTOR_SIZE` is in bytes.